### PR TITLE
Move Network XRD into Cluster

### LINKED ISSUE #25: Should the `CompositeNetwork` XRD offer a claim?
```console
$ kubectl get xrd
NAME                                                         ESTABLISHED   OFFERED   AGE
compositeclusters.aws.platformref.crossplane.io              True          True      28m
compositenetworks.aws.platformref.crossplane.io              True          True      28m
compositepostgresqlinstances.aws.platformref.crossplane.io   True          True      28m
eks.aws.platformref.crossplane.io                            True                    28m
services.aws.platformref.crossplane.io                       True                    28m
```

VPC networks are typically the canonical example of the kind of infrastructure platform operators want to manage, but platform consumers probably don't want to manage. Do we really want consumers managing VPC networks?

### PR DESCRIPTION
**Pull request by garreeoke**
### Description of your changes

Changes made to follow platform-ref-azure and incorporate the network fabric in to the creation of the cluster.

Fixes #25 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

Tested on local cluster and creating the cluster and rds instances.
Tested on hosted UBC cluster using CLI and UBC UI.

### CODE REVIEW

**Comment by jbw976** [comment]
@garreeoke I updated the title to better capture the intent of this PR for a public repo/audience, then added a link to the issue that this PR addresses into the body :)

File: README.md [hunk 1]
```diff
@@ -191,21 +208,9 @@ kubectl get claim
 kubectl get composite
 kubectl get managed
 ```
+>_Note: It may take some time for all managed resources to create successfully. Please reconcile using the AWS console._
```
Comment (jbw976) [inline README.md:211]: What is the intent of the messaging to "Please reconcile using the AWS console"? Ae we trying to say something like "You can also check the progress of these resources in the AWS console"?

File: cluster/definition.yaml [hunk 2]
```diff
@@ -82,15 +82,15 @@ metadata:
           validation:
           - required: false
 spec:
+  group: aws.platformref.crossplane.io
```
Comment (jbw976) [inline cluster/definition.yaml:85]: have you tested these changes in the UbC console also? I suspect the UI metadata (e.g. line 59 above) would also need to be updated to work OK

File: cluster/eks/composition.yaml [hunk 3]
```diff
@@ -35,7 +35,6 @@ spec:
                     }
                 ]
               }
-      name: controlplaneRole
```
Comment (jbw976) [inline cluster/eks/composition.yaml:38]: i don't think these base resource name fields should be removed - they were added in #63 as part of the effort to adopt the latest best practices in compositions

this comment applies to other composition files in this PR too

File: cluster/network/definition.yaml [hunk 4]
```diff
@@ -33,13 +33,13 @@ metadata:
           - required: true
             customError: Cluster ref is required.
 spec:
+  group: aws.platformref.crossplane.io
+  names:
+    kind: XNetwork
+    plural: xnetworks
   claimNames:
```
Comment (jbw976) [inline cluster/network/definition.yaml:40]: we should probably remove this claim/offering from this network XR - there isn't a use case for users creating namespaced instances of Networks - Networks will instead be created as part of other XRs

File: database/postgres/definition.yaml [hunk 5]
```diff
@@ -80,6 +83,5 @@ spec:
                     - id
                 required:
                   - storageGB
-                  - networkRef
```
Comment (jbw976) [inline database/postgres/definition.yaml:83]: should clusterRef be a required param now?

**Review by jbw976 — CHANGES_REQUESTED** [review]
This is great @garreeoke, thanks for tackling this improvement to the experience of this AWS reference platform! Let me know if you want to sync/chat on any of my suggestions for further clarity :)

File: README.md (hunk 1 above)
Comment (garreeoke) [reply README.md:211]: Yes, that was the original intent. It makes sense just to remove it.

File: cluster/eks/composition.yaml (hunk 3 above)
Comment (jbw976) [reply cluster/eks/composition.yaml]: awesome @garreeoke - can you give me a ping (@ mention me here) when this PR is ready to take another review pass on? thank you!

File: cluster/definition.yaml (hunk 2 above)
Comment (garreeoke) [reply cluster/definition.yaml:85]: @jbw976 I tested according to the README with a hosted cluster. What do I need to do to test in UBC?

**Comment by garreeoke** [comment]
@upbound/customers-community-squad I have tested this following the README. What additional steps would need to be done to test in UBC?

File: cluster/definition.yaml (hunk 2 above)
Comment (garreeoke) [reply cluster/definition.yaml:72]: I did test it in UBC and made some changes. This should be ready for review.

**Review by jbw976 — APPROVED** [review]

**Pull request merged**